### PR TITLE
Fixing bugs in the Lollipop Option in the Bar Chart dialog

### LINKED ISSUE #7295: Reopening the lollipop option on the bar chart dialogue
@N-thony I am re-opening this, because I can't get a good lollipop chart.  This is work you did earlier.  And I suggest these are needed, because they are used to analyse the R-Instat evaluation data.  You may wish to work with one of the interns, because there is a general feature also to correct.  
Lollipop first.  I'll sort out some data and add it here.  There seem to be 2 problems - at least!
a) The code gives not just a lollipop, but also a bar chart.  The geom lollipop is instead of the bar chart.  
b) When you use lollipop the button changes to Lollipop Options.  But it still gives bar chart.  It should give Lollipop.

Also, cosmetic options on this dialogue.  It currently looks a bit overwhelming.  So:
c) Make the first Reorder label into a checkbox.  Default unchecked. The control is only visible when checked.
d) I think the 2nd Reorder, and also Position are only when the By Factor (Optional): is not empty.  So make them not visible until that is filled.

There is a related issue on the plotting geoms.  But that will be separate and should involve an intern.  A careful one!

[R GUI Comparisons.zip](https://github.com/africanmathsinitiative/R-Instat/files/8237409/R.GUI.Comparisons.zip)

### PR DESCRIPTION
**Pull request by anastasia-mbithe**
This fixes #7295
@rdstern @N-thony, the Lollipop function does not include the bar chart function, so it only gives the Lollipop graph. The Lollipop options use geom_lollipop now. I have also added a checkbox for the first Reorder under rdoValue, and linked the second reorder and Position with the By Factor receiver.

### CODE REVIEW

**Review by rdstern — APPROVED** [review]
Excellent

File: instat/dlgBarAndPieChart.vb [hunk 1]
```diff
@@ -397,13 +400,13 @@ Public Class dlgBarAndPieChart
         bResetBarLayerSubdialog = True
 
         ucrInputAddReorder.SetText(strNone)
-        ucrInputAddReorder.bUpdateRCodeFromControl = True
+        'ucrInputAddReorder.bUpdateRCodeFromControl = True
```
Comment (Ivanluv) [inline instat/dlgBarAndPieChart.vb:403]: Could you remove the commented section

File: instat/dlgBarAndPieChart.vb [hunk 2]
```diff
@@ -397,13 +400,13 @@ Public Class dlgBarAndPieChart
...
 
         ucrInputReorderX.SetText(strNone)
         ucrInputReorderX.bUpdateRCodeFromControl = True
 
         ucrInputReorderValue.SetText(strNone)
-        ucrInputReorderValue.bUpdateRCodeFromControl = True
+        'ucrInputReorderValue.bUpdateRCodeFromControl = True
```
Comment (Ivanluv) [inline instat/dlgBarAndPieChart.vb:409]: Remove this too

File: instat/dlgBarAndPieChart.vb [hunk 3]
```diff
@@ -456,10 +459,10 @@ Public Class dlgBarAndPieChart
         clsForecatsInfreqValue.SetRCommand("fct_infreq")
 
         clsReorderFunctionValue.SetRCommand("reorder")
-        clsReorderFunctionValue.SetRCommand("reorder")
+        'clsReorderFunctionValue.SetRCommand("reorder")
```
Comment (Ivanluv) [inline instat/dlgBarAndPieChart.vb:462]: Remove

File: instat/dlgBarAndPieChart.vb [hunk 4]
```diff
@@ -694,14 +703,22 @@ Public Class dlgBarAndPieChart
     Private Sub SetDialogOptions()
         If rdoValue.Checked Or rdoFrequency.Checked Then
             clsRggplotFunction.AddParameter("mapping", clsRFunctionParameter:=clsBarAesFunction, iPosition:=1)
-            cmdBarChartOptions.Visible = True
+            'cmdBarChartOptions.Visible = True
```
Comment (Ivanluv) [inline instat/dlgBarAndPieChart.vb:706]: Remove

File: instat/dlgBarAndPieChart.vb [hunk 5]
```diff
@@ -694,14 +699,18 @@ Public Class dlgBarAndPieChart
...
             If Not ucrSaveBar.bUserTyped Then
                 ucrSaveBar.SetPrefix("bar_plot")
             End If
             ucrVariablesAsFactorForBarChart.RemoveIncludedMetadataProperty("class")
             ucrVariablesAsFactorForBarChart.strSelectorHeading = "Variables"
+            If Not ucrReceiverByFactor.IsEmpty() Then
+                ucrInputBarChartPositions.Visible = True
+            Else
+                ucrInputBarChartPositions.Visible = False
+            End If
```
Comment (lloyddewit) [inline instat/dlgBarAndPieChart.vb:713]:
```suggestion
            ucrInputBarChartPositions.Visible =  Not ucrReceiverByFactor.IsEmpty()
```

File: instat/dlgBarAndPieChart.vb [hunk 6]
```diff
@@ -710,25 +719,29 @@ Public Class dlgBarAndPieChart
...
-            ucrChkLollipop.Checked = False
             If ucrVariablesAsFactorForBarChart.bSingleVariable Then
                 ucrInputReorderX.Visible = True
-                ucrInputAddReorder.Visible = True
+                If Not ucrReceiverByFactor.IsEmpty() Then
+                    ucrInputAddReorder.Visible = True
+                Else
+                    ucrInputAddReorder.Visible = False
+                    ucrInputAddReorder.SetText(strNone)
+                End If
```
Comment (lloyddewit) [inline instat/dlgBarAndPieChart.vb:729]:
```suggestion
                ucrInputAddReorder.Visible = Not ucrReceiverByFactor.IsEmpty()
                If Not ucrInputAddReorder.Visible Then
                    ucrInputAddReorder.SetText(strNone)
                End If
```

File: instat/dlgBarAndPieChart.vb [hunk 7]
```diff
@@ -710,25 +719,29 @@ Public Class dlgBarAndPieChart
...
         ElseIf rdoValue.Checked Then
             If ucrVariablesAsFactorForBarChart.bSingleVariable Then
-                ucrInputReorderValue.Visible = True
-                ucrInputAddReorder.Visible = True
+                ucrChkReorderValue.Visible = True
+                If Not ucrReceiverByFactor.IsEmpty() Then
+                    ucrInputAddReorder.Visible = True
+                Else
+                    ucrInputAddReorder.Visible = False
+                    ucrInputAddReorder.SetText(strNone)
```
Comment (lloyddewit) [inline instat/dlgBarAndPieChart.vb:741]: See suggestion above

File: instat/dlgBarAndPieChart.vb [hunk 8]
```diff
@@ -877,8 +891,9 @@ Public Class dlgBarAndPieChart
         ChangeButtonsText()
     End Sub
 
-    Private Sub ucrPnlOptions_ControlValueChanged() Handles ucrPnlOptions.ControlValueChanged, ucrVariablesAsFactorForBarChart.ControlValueChanged, ucrReceiverX.ControlValueChanged, ucrReceiverByFactor.ControlValueChanged, ucrReceiverLabel.ControlValueChanged, ucrChkAddLabelsText.ControlValueChanged,
-    ucrInputReorderX.ControlValueChanged, ucrInputAddReorder.ControlValueChanged, ucrInputReorderValue.ControlValueChanged, ucrNudMaxSize.ControlValueChanged, ucrChkIncreaseSize.ControlValueChanged
+    Private Sub ucrPnlOptions_ControlValueChanged() Handles ucrPnlOptions.ControlValueChanged, ucrVariablesAsFactorForBarChart.ControlValueChanged, ucrReceiverX.ControlValueChanged, ucrReceiverByFactor.ControlValueChanged, ucrReceiverLabel.ControlValueChanged,
+        ucrChkAddLabelsText.ControlValueChanged, ucrChkReorderValue.ControlValueChanged,
+    ucrInputReorderX.ControlValueChanged, ucrInputAddReorder.ControlValueChanged, ucrInputReorderValue.ControlValueChanged, ucrNudMaxSize.ControlValueChanged, ucrChkIncreaseSize.ControlValueChanged, ucrChkLollipop.ControlValueChanged
```
Comment (lloyddewit) [inline instat/dlgBarAndPieChart.vb:896]: Add extra line breaks (max line length should be 100 columns), and fix indents

File: instat/dlgBarAndPieChart.vb [hunk 9]
```diff
@@ -900,7 +914,7 @@ Public Class dlgBarAndPieChart
         End If
     End Sub
 
-    Private Sub ucrChkBacktoback_ControlValueChanged(ucrChangedControl As ucrCore) Handles ucrChkBacktoback.ControlValueChanged, ucrChkLollipop.ControlValueChanged, ucrVariablesAsFactorForBarChart.ControlValueChanged, ucrReceiverByFactor.ControlValueChanged, ucrReceiverX.ControlValueChanged, ucrPnlOptions.ControlValueChanged
+    Private Sub ucrChkBacktoback_ControlValueChanged(ucrChangedControl As ucrCore) Handles ucrChkBacktoback.ControlValueChanged, ucrVariablesAsFactorForBarChart.ControlValueChanged, ucrReceiverX.ControlValueChanged
```
Comment (lloyddewit) [inline instat/dlgBarAndPieChart.vb:917]: add extra line breaks

**Review by lloyddewit — CHANGES_REQUESTED** [review]
@Ivanluv Thank you for the peer review.
@anastasia-mbithe looks good, just a few more questions/comments

**Comment by anastasia-mbithe** [comment]
@lloyddewit, Can you kindly have a look at this.

**Review by lloyddewit — APPROVED** [review]
@anastasia-mbithe thanks for the changes, looks good!

**Comment by lloyddewit** [comment]
@rdstern just a few changes since your last approval, if you can test/approve again, then we can merge.
Thanks

**Review by rdstern — APPROVED** [review]
Looks very good now.

**Pull request merged**